AlfrescoApp/Config/Service/AlfrescoClientBasedConfigService.m

```diff
@@ -225,25 +226,57 @@ - (AlfrescoRequest *)initializeInternalStateWithCompletionBlock:(AlfrescoBOOLCom
                                             
                                             NSError *updateAttributesError = nil;
                                             NSDictionary *updatedAttributes = @{NSFileModificationDate : configNode.modifiedAt};
-                                            [[NSFileManager defaultManager] setAttributes:updatedAttributes ofItemAtPath:completeFileConfigPath error:&updateAttributesError];
+                                            [[NSFileManager defaultManager] setAttributes:updatedAttributes ofItemAtPath:temporaryFileConfigPath error:&updateAttributesError];
                                             
                                             if (updateAttributesError)
                                             {
                                                 AlfrescoLogError(@"Unable to set the attributes: %@ on file at path: %@", updatedAttributes, completeFileConfigPath);
                                             }
                                             
                                             // process the JSON
-                                            [self processJSONData:[NSData dataWithContentsOfFile:completeFileConfigPath] completionBlock:^(BOOL succeeded, NSError *error) {
+                                            [self processJSONData:[NSData dataWithContentsOfFile:temporaryFileConfigPath] completionBlock:^(BOOL succeeded, NSError *error) {
                                                 // Notify processing is complete
                                                 if (succeeded)
                                                 {
-                                                    [[NSNotificationCenter defaultCenter] postNotificationName:kAlfrescoConfigNewConfigRetrievedFromServerNotification object:nil];
+                                                    // delete old config file
+                                                    if ([[AlfrescoFileManager sharedManager] fileExistsAtPath:completeFileConfigPath])
+                                                    {
+                                                        NSError *deleteError;
+                                                        [[AlfrescoFileManager sharedManager] removeItemAtPath:completeFileConfigPath error:&deleteError];
```

Comment:
Minor observation: No need to allocate and pass a variable reference if the error is going to be ignored anyway